jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/ObjectRule.java

```diff
@@ -425,11 +425,30 @@ private void addInterfaces(JDefinedClass jclass, JsonNode javaInterfaces) {
     }
 
     private String getClassName(String nodeName, JPackage _package) {
-        String className = ruleFactory.getNameHelper().replaceIllegalCharacters(capitalize(nodeName));
+        String prefix = ruleFactory.getGenerationConfig().getClassNamePrefix();
+        String suffix = ruleFactory.getGenerationConfig().getClassNameSuffix();
+        String fullNodeName = createFullNodeName(nodeName, prefix, suffix);
+
+        //Note that only the first letter of the fullNodeName will be capitalized not the node name
+        //so prefix + nodeName becomes PrefixnodeName
```

Comment:
Agreed. Without capitalizing nodeName, I think the majority of people will be disappointed with how this behaves.